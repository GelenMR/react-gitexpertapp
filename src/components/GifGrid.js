import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
// import { getGifs } from '../helpers/getGifs';

export const GifGrid = ( { category } ) => {

    /* const [ images, setImages ] = useState( [] );

    useEffect( () => {
        getGifs( category )
            .then( setImages );
    }, [ category ])

    const getGifs = async() => {
        const url = `https://api.giphy.com/v1/gifs/search?q=${ encodeURI( category ) }&limit=10&api_key=a3z2wabmEvP8aac1V3Ys88TGexqpqvvz`;
        const resp = await fetch(url);
        const { data } = await resp.json();
        // console.log(data);
        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        console.log(gifs);
        setImages(gifs);
    }
    // getGifs(); */

    const { data:images, loading } = useFetchGifs( category );
    console.log(loading);
    console.log(images);

    return (
        <>
            <h3>{ category }</h3>
            {/* { loading ? 'Cargando...' : 'Data Cargada!' } */}
            <div className="animate__animated animate__flash">
                {loading && 'Cargando...'}
            </div>
            <div className="card-grid">
                {/* {images.map( ( img ) => ( */}
                {images.map( img  => (
                    // <li key={ id }>{ title }</li>
                    <GifGridItem 
                        key={ img.id }
                        { ...img }// img={ img }
                    />
                ))}
            </div>
        </>
    )
}