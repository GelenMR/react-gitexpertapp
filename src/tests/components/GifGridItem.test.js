import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en componente <GifGridItem />', () => {
    const title = 'Titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem title={title} url={url} /> );

    test('Debe de mostrar en componente correctamente', () => {
       
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe de tener un parrafo con el titulo', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    });
    
    test('Debe de tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        console.log(img.html());
        console.log(img.props());
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('Debe de tener animate__zoomIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect( className.includes('animate__zoomIn') ).toBe( true );
    })
    
    
});
