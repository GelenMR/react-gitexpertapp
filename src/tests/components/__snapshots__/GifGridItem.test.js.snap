// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifGridItem /> Debe de mostrar en componente correctamente 1`] = `
<div
  className="card animate__animated animate__zoomIn"
>
  <img
    alt="Titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Titulo
  </p>
</div>
`;
